fs-objects.go

```diff
@@ -143,7 +145,73 @@ func (fs fsObjects) GetObjectInfo(bucket, object string) (ObjectInfo, error) {
 
 // PutObject - create an object.
 func (fs fsObjects) PutObject(bucket string, object string, size int64, data io.Reader, metadata map[string]string) (string, error) {
-	return putObjectCommon(fs.storage, bucket, object, size, data, metadata)
+	// Verify if bucket is valid.
+	if !IsValidBucketName(bucket) {
+		return "", BucketNameInvalid{Bucket: bucket}
+	}
+	// Check whether the bucket exists.
+	if !isBucketExist(fs.storage, bucket) {
+		return "", BucketNotFound{Bucket: bucket}
+	}
+	if !IsValidObjectName(object) {
+		return "", ObjectNameInvalid{
+			Bucket: bucket,
+			Object: object,
+		}
+	}
+
+	fileWriter, err := fs.storage.CreateFile(bucket, object)
```

Comment:
PutObject diverged because of the following requirement in XL:
```
+	// check if an object is present as one of the parent dir.
 +	if err = xl.parentDirIsObject(bucket, path.Dir(object)); err != nil {
 +		return "", toObjectErr(err, bucket, object)
 +	}
 +
 +	// Delete if an object already exists.
 +	// FIXME: rename it to tmp file and delete only after
 +	// the newly uploaded file is renamed from tmp location to
 +	// the original location.
 +	err = xl.deleteObject(bucket, object)
 +	if err != nil && err != errFileNotFound {
 +		return "", toObjectErr(err, bucket, object)
 +	}
```